src/slurmrestd_ops.py

```diff
@@ -0,0 +1,333 @@
+        return slurmrestd.version.number
+
+
+class SlurmrestdOpsManager:
+    """SlurmrestdOpsManager."""
+
+    def __init__(self):
+        """Init the class."""
+        self._slurm_conf_dir = Path("/etc/slurm")
+        self._slurm_conf_path = self._slurm_conf_dir / "slurm.conf"
+
+    def install(self) -> bool:
+        """Install slurmrestd to the system."""
+        if Slurmrestd().install() is not True:
+            return False
+
+        self._create_slurmrestd_user_group()
+
+        systemd.service_enable(self._munged_systemd_service)
+        systemd.service_enable(self._slurmrestd_systemd_service)
+
+        systemd.service_stop(self._munged_systemd_service)
+
+        if not self._slurm_conf_dir.exists():
+            self._slurm_conf_dir.mkdir()
+
+        user = f"{self._slurmrestd_user_name}:{self._slurmrestd_group_name}"
+        subprocess.call(["chown", "-R", user, self._slurm_conf_dir])
+
+        self.setup_slurmrestd_systemd_unit()
+        return True
+
+    def setup_slurmrestd_systemd_unit(self):
+        """Replace default systemd unit.
+
+        The default service unit uses a local Unix socket for connection, we
+        explicitly disable it and expose the port 6820.
+        """
+        logger.debug("## Replacing slurmrestd.service")
+        target = Path("/usr/lib/systemd/system/slurmrestd.service")
+        source = TEMPLATE_DIR / self._slurmrestd_systemd_service
+
+        if target.exists():
+            target.unlink()
+
+        target.write_text(source.read_text())
+        try:
+            subprocess.call(["systemctl", "daemon-reload"])
+        except subprocess.CalledProcessError as e:
+            logger.error(e)
+            raise (e)
+
+    def write_slurm_conf(self, slurm_conf: str) -> None:
+        """Render the context to a template, adding in common configs."""
+        target = self._slurm_conf_path
+
+        if target.exists():
+            target.unlink()
+
+        target.write_text(slurm_conf)
+
+        user_group = f"{self._slurmrestd_user_name}:{self._slurmrestd_group_name}"
+        subprocess.call(["chown", user_group, target])
+
+    def write_munge_key(self, munge_key) -> None:
+        """Base64 decode and write the munge key."""
+        key = b64decode(munge_key.encode())
+        self._munge_key_path.write_bytes(key)
+
+    def _is_active_munged(self) -> bool:
+        munge = self._munged_systemd_service
+        try:
+            status = subprocess.check_output(f"systemctl is-active {munge}", shell=True)
+            status_str = status.decode().strip()
+            if "active" in status_str:
+                logger.debug("#### Munge daemon active")
+                return True
+            else:
+                logger.error(f"## Munge not running: {status_str}")
+                return False
+        except subprocess.CalledProcessError as e:
+            logger.error(f"## Error querring munged - {e}")
+            return False
```

Comment:
Made a similar comment on the slurmd operator IIRC, but I don't think we need all this logic here as it can be superseded by the `service_running(...)` function in the `systemd` charm library:

```suggestion
        return systemd.service_running("munged")
```

In fact, you might not even need this method as you can just use the `service_running(...)` function to get the current state of the `munged` service.